src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -414,10 +414,9 @@ public Collection GetDefaultCollection()
                         WriteLog.To.Database.W(Tag, $"The none-recoverable default collection is now deleted from database {ToString()}.");
                     }
                 }
+                
+                return _defaultCollection == null || !_defaultCollection.IsValid ? null : _defaultCollection;
```

Comment:
Simplified version is `return _defaultCollection?.IsValid == true ? _defaultCollection : null`